### DISCUSSION TITLE
Right panel no longer persists in the same way between rooms

### DISCUSSION THREAD

**Discussion by turt2live** [element-hq/element-meta#94] [General]
Previously it'd try and keep the same kind of panel open between rooms, but some recent change has meant it is closed by default and is per-room. This leads to jarring timeline behaviour as the width changes between rooms

**Comment by jryans** [comment]
Yes, it is currently expected to be per-room after implementing the threads / maximised widgets [right panel navigation changes](https://github.com/vector-im/element-web/issues/19660#issuecomment-971574851).

It does have the downside that the width may change now, yes.

**Comment by SimonBrandner** [comment]
Personally, I find it quite annoying to have to open the right panel in every room by hand

**Comment by toger5** [comment]
I was running it for some weeks now and got really used to it. If you have set the right panel in most rooms to what you want it to be the right panel in each room it becomes nicer to use. I can see how the transition is not that pleasant. We could consider a default right panel and an additional global right panel open/close state?


But this is definitely a complex UX decision. I am biased of course. But still I have an insight in the arguments made for that change so I can list them for the discussion:
On the one hand it is really beneficial to be able to have the meter list open in a room where you might want to check the members often (bigger room) or not have any panel in a private chat where the member panel content is obvious. Or pinned messages in a room where they are used for annocements or a chat in a room where there is a maximised widget and nothing where the room is only used for the widget and not really to interact in chat ... (you get the point in some situations it is beneficial to have it per room)

The other side of already got mentioned here: opening the panel for each room is annoying because the default is: no right panel.
And that the state changes per room leading to a different timeline width in different rooms.

With threads and widgets the trend seem to be, that there is more per room information next to the timeline. So having the right panel state as part of the room seems reasonable to me.

A more sophisticated solution could be:
Local right panel phases: timeline & thread & right panel widgets.
Global right panel state: all the others.
We could only store it per room if one of the local phases is open. And otherwise just show the current global phase (and follow the global open state)

**Comment by SimonBrandner** [comment]
For me personally, a default opened state would probably be enough

**Comment by kittykat** [comment]
I'm going to move this issue to a Discussion in element-meta as this is working as intended and the question is around improving default behaviour

**Comment by ara4n** [comment]
So on one hand, I love that the RightPanel now remembers its state per room.  On the other hand, it's super frustrating that it's hidden by default - i literally spend all my life manually opening it.  Obviously I'll eventually get to the point where i've manually had to open the RightPanel for all my rooms and won't have to do it any more, but it's very tedious to get there.

Please can we at least have an option to show it by default (defaulting to memberlist) - but still remember per-room which panel it's showing?

**Comment by jryans** [comment]
Thanks for all the feedback so far! 😄 I'm checking with Design on the best way to proceed here.

**Reply by daniellekirkwood** [reply]
There might also be some analytics events we could use to make a proposal based on current usage :)

**Reply by jryans** [reply]
Design is continuing to think through this problem, but there are several nuances to untangle, so may take a bit more time.

**Reply by turt2live** [reply]
While we wait for design, can we get the setting proposed by Matthew up above? My frustration with the behaviour is starting to convert into anger.

**Reply by babolivier** [reply]
Or have a setting to allow to switch between per-room and global to the app? That would prevent Element from frustrating me every time I switch rooms.

**Reply by jryans** [reply]
I don't think a setting that allows global to the app makes sense with the advent of threads... It would revive the poor UX of a thread from room A following you around app when you switch to room B, which is one of the main reasons a change was made.

Something like the setting Matthew proposes is perhaps the quickest option, but I'd still like to get some Design approval first. I have asked again specifically about something like this, will let you know.

**Reply by babolivier** [reply]
I would say threads should probably be separated from the right panel (not sure where to move them to, though). Both their semantics and navigation are very clearly separated from the rest of the right panel imo, and should not share the space with it.

**Reply by jryans** [reply]
Perhaps so, but that feels like a much longer term rework of the info hierarchy of the app.

**Reply by babolivier** [reply]
Yes, sorry if I make it sound like I think it's a quick job. I'm mostly trying that come up with a theoretical design that wouldn't make me frustrated by the unstable right panel, without considering the effort on purpose.

**Comment by babolivier** [comment]
Copying some feedback from https://github.com/vector-im/element-web/issues/20602 (which I opened before knowing this discussion was a thing, closed now), which summarises my experience after a few days of using the new behaviour:

I would expect the right panel to stay the same between room changes. If I e.g. open my list of notifications, or change my view of the room to display a member list, I expect this layout change to persist when changing rooms. Otherwise I need to go through all of my rooms to change the layout on each of them every time I need something different.

I find the current behaviour (changes are saved for a room rather than for the whole app) overly confusing, because it does not match the behaviour of the other panels (e.g. left panels don't change when changing rooms), doesn't seem to match the semantics of some of the right panel sections (e.g. the notifications one, which spans across multiple rooms), and I very often end up switching back to a room hours later and having to figure out why Element shows me a specific section, or the profile of a specific user, instead of my usual layout. Or why I have the member list open in one room and closed in another. Most often it just makes me waste time by making me manually set the right panel to the right state every time I switch rooms.

With the current behaviour I'd rather not be using the right panel at all but I can't even do that without manually going through each room I'm in to close it.

**Reply by toger5** [reply]
The notification panel indeed feels rather out of place It truly is not a per room panel. And i think it is going to be dropped eventually and replaced by sth better.
The other panels on the other hand are per room and dependent on which features you use having the thread panel open in a room where you dont have threads because you were reading a thread in the previous room can be equally annoying then not having the room member list open. In both scenarios you are forced to manually change it to what you want to see.
Which situation occurs more often is very user dependent. So every feedback is really important to figure out what happens the most.

**Reply by babolivier** [reply]
>  The other panels on the other hand are per room and dependent on which features you use having the thread panel open in a room where you dont have threads because you were reading a thread in the previous room can be equally annoying then not having the room member list open. In both scenarios you are forced to manually change it to what you want to see.

I don't fully agree. To me, while the information in the panel does depend on the room, the panel itself (its presence and configuration (i.e. which view it's on)) is part of the whole app's interface. This is why it feels very weird and counter-intuitive to me to see it disappear or change between rooms.

**Reply by babolivier** [reply]
As a follow-up, I've tried the new labs flag and it didn't change my experience that much, since most of my frustration comes from the fact that the right panel changes views across rooms, so all that means to me is ever few room switches I have to waste a bit of time to reset the right panel to the view it had in the room I was previously viewing.

**Comment by daniellekirkwood** [comment]
Thanks for the feedback! This discussion is valuable and while we work on collecting more data we have a temporary solution for y'all. 

**We will add a Labs flag that, when on, will cause the right panel to remain open in all rooms. Each room will remember the view it was last showing and present that to the user. If the room's right panel was closed, or it hasn't yet been opened it will default to show the "People" list.** 

Further feedback on this solution is welcome, but this is our starting point. 
We're seeking to learn from Labs, Feedback, and maybe even PostHog 🙂
This GH Discussion will remain open in order to gather our learnings and suggest this issue as an Epic for Quarterly Planning in due time.

👀  @jryans & @janogarcia

**Reply by jryans** [reply]
Thanks @daniellekirkwood! 😄 I'll work on implementing this now.

**Reply by babolivier** [reply]
Thanks @daniellekirkwood! I'd love to see a labs flag that makes room __not__ remember the view it was last showing, but I don't assume that would be possible?

**Reply by toger5** [reply]
@babolivier this actually would be super simple to add. The right panel store has a property called `currentRoom: string`. If we disable all situations in which we set the currentRoom variable and just leave it as "noRoom" or "" (empty string) the right panel would think it is always showing one room and only storing the history of that one room.

(The setCard function takes an optional argument for a room id that would then also need to be ignored if yhe flag is activated)

**Reply by toger5** [reply]
Be aware, that this would put both: threads and the right panel chat in the awkward situation, that they can be open in rooms they don't belong to

**Comment by daniellekirkwood** [comment]
Temporary solution issue: https://github.com/vector-im/element-web/issues/20666

**Reply by jryans** [reply]
This flag is now available on develop. Please look for "Right panel stays open (defaults to room member list)" in the Rooms section of Labs.
